core/src/main/java/org/elasticsearch/search/internal/SearchContext.java

```diff
@@ -360,6 +360,8 @@ public void clearReleasables(Lifetime lifetime) {
 
     public abstract Counter timeEstimateCounter();
 
+    public abstract boolean hasFetchSubPhaseContext(FetchSubPhase.ContextFactory contextFactory);
```

Comment:
Since the implementation of this seems to be a check in the hash map, does it make sense to have it as a separate method instead of annotating `getFetchSubPhaseContext` with `@Nullable`?